### ISSUE TITLE
How to get the prediction Output after model.fit(DataSet) ?

### ISSUE THREAD

**Issue by DavidScisly** [deeplearning4j/deeplearning4j#7136]
#### Issue Description

Please describe your issue, along with:

    * **expected behavior**
      After training the model.fit(DataSet(nd_input, nd_output), i expected to get an output value for each row from nd_input

    * **encountered behavior**
      output is always 0.00

#### Version Information

Please indicate relevant versions, including, if relevant:

    * **Deeplearning4j version**
      0.9.1

    * **platform information (OS, etc)**
      windows 10, intellij community edition

#### Tested Programm

    * **Code**
      public class NNdataDL4J {
      private Data data;
      private MinMax MM;
      //private NeuralNetwork neuralNet;
      private MultiLayerNetwork neuralNet;
      private DataSet trainSet, testSet;
      private int inputNeurons, hiddenNeurons, outputNeurons = 1;
      private double learningrate, maxerror;
      private double[] norm;
      private INDArray nd_input, nd_sliced, nd_output, nd_test;
      public NNdataDL4J(double lr, double mr, int in, int on, Data data){
      this.learningrate   = lr;
      this.maxerror       = mr;
      this.inputNeurons   = in;
      this.hiddenNeurons  = on;
      this.data           = data;
      this.MM             = data.getMM();
      ```
            int seed = 123;
            MultiLayerConfiguration mlConf = new NeuralNetConfiguration.Builder()
                    .seed(seed)
                    .iterations(100)
                    .learningRate(learningrate)
                    .activation(Activation.RELU)
                    .optimizationAlgo(OptimizationAlgorithm.STOCHASTIC_GRADIENT_DESCENT)
                    .list()
                    .layer(0, new DenseLayer.Builder()
                            .nIn(inputNeurons)
                            .nOut(hiddenNeurons)
                            .weightInit(WeightInit.XAVIER)
                            .build()
                    ).layer(1, new OutputLayer.Builder()
                            .lossFunction(LossFunctions.LossFunction.NEGATIVELOGLIKELIHOOD)
                            .weightInit(WeightInit.XAVIER)
                            .nIn(hiddenNeurons)
                            .nOut(outputNeurons)
                            .build()
                    )
            .pretrain(true).backprop(false).build();
      ```
      //            System.out.println(mlConf.toJson());
      neuralNet = new MultiLayerNetwork(mlConf);
      neuralNet.init();
      neuralNet.setListeners(new ScoreIterationListener(1));
      }
      public void trainSet(){
      norm = data.get(1);
      nd_output   = Nd4j.create(norm.length-inputNeurons,1);
      nd_input    = Nd4j.create(norm.length-inputNeurons, inputNeurons);
      for(int i=0; i<nd_input.size(0); ++i){
      double[] dtemp      = new double[inputNeurons];
      System.arraycopy(norm, i, dtemp, 0, dtemp.length);
      INDArray nd_temp    = Nd4j.create(dtemp, new int[]{1,inputNeurons});
      nd_input.putRow(i, nd_temp);
      }
      ```
        System.out.println("InputValues: ");
        for(int i=0; i < nd_input.size(0); ++i)
            System.out.println("Input, row: "+ i + nd_input.getRow(i));
        System.out.println("Output values: ");
      
        trainSet = new DataSet(nd_input, nd_output);
        neuralNet.fit(trainSet);
        System.out.println("Output: " + neuralNet.output(nd_input));
      ```
      }

    * **Output**
      InputValues:
      Input, row: 0[0.48,  0.49,  0.49,  0.45]
      Input, row: 1[0.49,  0.49,  0.45,  0.42]
      Input, row: 2[0.49,  0.45,  0.42,  0.46]
      Input, row: 3[0.45,  0.42,  0.46,  0.44]
      Input, row: 4[0.42,  0.46,  0.44,  0.44]
      Input, row: 5[0.46,  0.44,  0.44,  0.46]
      Input, row: 6[0.44,  0.44,  0.46,  0.43]
      Input, row: 7[0.44,  0.46,  0.43,  0.45]
      Input, row: 8[0.46,  0.43,  0.45,  0.45]
      Input, row: 9[0.43,  0.45,  0.45,  0.48]
      Input, row: 10[0.45,  0.45,  0.48,  0.47]
      Input, row: 11[0.45,  0.48,  0.47,  0.50]
      Input, row: 12[0.48,  0.47,  0.50,  0.51]
      Input, row: 13[0.47,  0.50,  0.51,  0.50]
      Input, row: 14[0.50,  0.51,  0.50,  0.52]
      Input, row: 15[0.51,  0.50,  0.52,  0.52]
      Input, row: 16[0.50,  0.52,  0.52,  0.53]
      Input, row: 17[0.52,  0.52,  0.53,  0.55]
      Input, row: 18[0.52,  0.53,  0.55,  0.56]
      Output values:
      Output: [0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00,  0.00]

    * **Questions**

    * * I'm not sure if i set the DataSet right. DataSet(features = nd_input(size: y rows, x columns), labels = nd_output(size: y rows, 1 column) ?

    * * any ideas how i get the output of the model for every input row?

    * * Why the Iteration Listener doesn't work?

    * * any other tipps?

**Comment by raver119** [comment]
1) Consider update to latest stable dl4h/nd4j version, which is 1.0.0-beta3

2) model.output() is the right way.

3) Your network is missing actual activation function for OutputLayer, so your default RELU is used.  So it looks a bit strange combined with your loss function :)

**Comment by DavidScisly** [comment]
thanks a lot @raver119 , it works now with activation function in outputlayer.
